pkg/security/podsecuritypolicy/util/util.go

```diff
@@ -182,8 +183,9 @@ func PSPAllowsHostVolumePath(psp *extensions.PodSecurityPolicy, hostPath string)
 		return true
 	}
 
+	path := filepath.Clean(hostPath)
```

Comment:
I remember that `hasPathPrefix()` has some optimizations for not allocated memory and re-use existing slice. I'm wondering if we use `Clean()`  here, will it lower our optimization tactics below?

Another question: would `filepath.Clean()` work for Windows-style paths? (I believe that answer is No.) And should we care about it at all?

Would it be easier to search for `..` and reject such paths without trying to normalize them?